File: backendAPI/SecureAPI.py
from flask import Flask, request, jsonify
from flask_limiter import Limiter
from flask_cors import CORS

app = Flask(__name__)
CORS(app)
limiter = Limiter(app, key_func=lambda: request.remote_addr, default_limits=["50 per minute"])

@app.route('/api/secure', methods=['POST'])
def secure():
    data = request.get_json()
    username = data.get('username')
    password = data.get('password')

    if not username or not password:
        return jsonify({'error': 'Missing fields'}), 400

    print(f"[SECURE] Login attempt by {username}")
    return jsonify({'success': True, 'message': 'Data accepted'})

if __name__ == '__main__':
    app.run(debug=True, port=5000)

File: backendAPI/SecureAPI.js
const express = require('express');
const cors = require('cors');
const helmet = require('helmet');
const rateLimit = require('express-rate-limit');
const logger = require('../utils/logger');
require('dotenv').config({ path: '../utils/config.env' });

const app = express();
const PORT = process.env.PORT || 5000;

app.use(cors({ origin: 'https://atomicv2.netlify.app/' }));
app.use(helmet());
app.use(express.json());

// Rate limiting
const limiter = rateLimit({ windowMs: 1 * 60 * 1000, max: 50 });
app.use(limiter);

// Endpoint
app.post('/api/secure', (req, res) => {
  const { username, password } = req.body;

  if (!username || !password) {
    logger.warn('Missing fields attempt');
    return res.status(400).json({ error: 'Missing fields' });
  }

  logger.log(`Login attempt by ${username}`);
  return res.status(200).json({ success: true, message: 'Data accepted' });
});

app.listen(PORT, () => {
  console.log(`🔒 SecureAPI.js listening on http://localhost:${PORT}`);
});
